pkg/products/grafana/reconciler.go

```diff
@@ -0,0 +1,320 @@
+						"-pass-basic-auth=false",
+						"-https-address=:9091",
+						"-http-address=",
+						"-email-domain=*",
+						"-upstream=http://localhost:3000",
+						"-openshift-sar={\"resource\":\"namespaces\",\"verb\":\"get\"}",
+						"-openshift-delegate-urls={\"/\":{\"resource\":\"namespaces\",\"verb\":\"get\"}}",
+						"-tls-cert=/etc/tls/private/tls.crt",
+						"-tls-key=/etc/tls/private/tls.key",
+						"-client-secret-file=/var/run/secrets/kubernetes.io/serviceaccount/token",
+						"-cookie-secret-file=/etc/proxy/secrets/session_secret",
+						"-openshift-service-account=grafana-serviceaccount",
+						"-openshift-ca=/etc/pki/tls/cert.pem",
+						"-openshift-ca=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
+						"-skip-auth-regex=^/metrics"},
+					Ports: []v1.ContainerPort{
+						{ContainerPort: 9091,
+							Name: "grafana-proxy"},
+					},
+				},
+			},
+			Secrets: []string{"grafana-k8s-tls", "grafana-k8s-proxy"},
+			Service: &grafanav1alpha1.GrafanaService{
+				Ports: []v1.ServicePort{
+					{Name: "grafana-proxy",
+						Port:     9091,
+						Protocol: v1.ProtocolTCP,
+					},
+				},
+				Annotations: annotations,
+			},
+			Ingress: &grafanav1alpha1.GrafanaIngress{
+				Enabled:     true,
+				TargetPort:  "grafana-proxy",
+				Termination: "reencrypt",
+			},
+			Client: &grafanav1alpha1.GrafanaClient{
+				PreferService: true,
+			},
+			Compat: &grafanav1alpha1.GrafanaCompat{
+				FixAnnotations: true,
+			},
+			ServiceAccount: &grafanav1alpha1.GrafanaServiceAccount{
+				Annotations: serviceAccountAnnotations,
+			},
+		},
+	}
+
+	status, err := controllerutil.CreateOrUpdate(ctx, client, grafana, func() error {
+		owner.AddIntegreatlyOwnerAnnotations(grafana, r.installation)
+
+		return nil
+	})
+
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, err
+	}
+
+	logrus.Info("Grafana Status: ", status)
+
+	// if there are no errors, the phase is complete
+	return integreatlyv1alpha1.PhaseCompleted, nil
+}
+
+func (r *Reconciler) reconcileSubscription(ctx context.Context, serverClient k8sclient.Client, inst *integreatlyv1alpha1.RHMI, productNamespace string, operatorNamespace string) (integreatlyv1alpha1.StatusPhase, error) {
+	r.logger.Info("reconciling subscription")
+
+	target := marketplace.Target{
+		Pkg:       constants.GrafanaSubscriptionName,
+		Namespace: operatorNamespace,
+		Channel:   marketplace.IntegreatlyChannel,
+	}
+	catalogSourceReconciler := marketplace.NewConfigMapCatalogSourceReconciler(
+		manifestPackage,
+		serverClient,
+		operatorNamespace,
+		marketplace.CatalogSourceName,
+	)
+	return r.Reconciler.ReconcileSubscription(
+		ctx,
+		target,
+		[]string{productNamespace},
+		r.preUpgradeBackupExecutor(),
+		serverClient,
+		catalogSourceReconciler,
+	)
+}
+
+func (r *Reconciler) preUpgradeBackupExecutor() backup.BackupExecutor {
+	return backup.NewNoopBackupExecutor()
+}
+
+// PopulateSessionProxySecret generates a session secret
+func PopulateSessionProxySecret() string {
```

Comment:
Does this need to be exported. I think it's used somewhere else as well. Is it exported there too? Could we combine or just have it internal.

Reply:
No real need but the other exported version is in application monitoring which I dont want to rely on always being there. I've stopped the export but left the function in.